boot/zephyr/pm.yml

```diff
@@ -0,0 +1,12 @@
+mcuboot:
+  placement:
+    before: [spm, app]
+mcuboot_partitions:
+  inside: [spm, app]
```

Comment:
What does `inside` do?

Reply:
I creates sub-partitions.  Note that the documentation will be done ASAP after code is in (hopefully that is OK).

For the case of MCUboot, it needs to have a concept of a partition which consists of everything that should be updated. This because it doesn't support multi image updates ATM.

When an app is built with SPM, those two images needs to be treated as one by mcuboot. This is what is represented here.

Reply:
If `inside` creates subpartitions, what does `subpartition` do?

Reply:
One level nesting is supported.

The second nested level only supports equal split of the parent partition.

It is done like this to support the MCUboot use case.

`inside` indicates where the listed `sub-partitions` will be placed.

Reply:
I updated the keyword to be `span` instead. See if that makes more sense.

Reply:
`Define a set of sub-partitions which spans over 'mcuboot_pad', 'spm' and 'app'`
Does that mean that `primary` and `secondary` partitions have `mcuboot_pad`, `spm` and `app` "subpartitions"?